woocommerce/woocommerce-ios
# Login: Site discovery on the empty store picker

Closes: #7468. Also closes #7349.


### Description

This PR adds 2 new buttons: "New To WooCommerce?" and "Enter your site address" on the site picker screen when no store connected to a user's WP.com account is found. This button navigates to a new screen for enter a site address, and from here an appropriate error message is displayed based on the returned site info.

To optimize the effort required for this task, I added a new flow in the WPAuthenticator library in https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/pull/666 to reuse the UI for the site address screen and also the check for the site info.

The message on the empty store picker is also updated, and the Jetpack instruction button is removed.

This PR also fixes #7349 by moving the "Continue" button to the top of the stack in the store picker.

### Testing instructions

Prerequisite: please make sure that you have a WP.com account that's not associated with any WP site or Woo store.
1. Log out of the app if necessary, and select Continue with WP.com.
2. Log in to your WP.com account. After the login succeeds, notice that the empty site picker is displayed.
3. Notice that the empty site picker no longer shows the message or CTA about Jetpack, and instead has a button for enter a site address.
4. Select the "New To WooCommerce" button, notice in Xcode console: `🔵 Tracked site_picker_new_to_woo_tapped, properties: [:]`.
5. Select the "Enter your site address" button, notice in Xcode console: `🔵 Tracked site_picker_enter_site_address_tapped, properties: [:]`. Notice that you are navigated to a new screen to enter site address.
6. Enter an invalid site address, notice that the screen shows an inline error message saying the address is invalid.
7. Enter the address of a non-existent site, notice that the screen shows an inline error message saying the site is not accessible.
8. Enter the address of a site without WP, notice that the No WP error screen is presented. Notice in Xcode console: `🔵 Tracked site_picker_site_discovery, properties: [AnyHashable("is_wpcom"): false, AnyHashable("has_wordpress"): false, AnyHashable("has_valid_jetpack"): false]`.
9. Enter the address of a WP.com site, notice that the Account Mismatch error screen is presented. Notice in Xcode console: `🔵 Tracked site_picker_site_discovery, properties: [AnyHashable("is_wpcom"): true, AnyHashable("has_wordpress"): true, AnyHashable("has_valid_jetpack"): <false or true based on whether the site is atomic>]`.
10. Enter the address of a self-hosted WP with Jetpack, notice that the Account Mismatched error screen is presented. Notice in Xcode console: `🔵 Tracked site_picker_site_discovery, properties: [AnyHashable("is_wpcom"): false, AnyHashable("has_wordpress"): true, AnyHashable("has_valid_jetpack"): true]`.
11. Enter the address of a self-hosted WP site without Jetpack, notice that the Jetpack error screen is presented. Notice in Xcode console: `🔵 Tracked site_picker_site_discovery, properties: [AnyHashable("is_wpcom"): false, AnyHashable("has_wordpress"): true, AnyHashable("has_valid_jetpack"): false]`.
12. Follow the instruction to install Jetpack, if the site doesn't have Woo, the No Woo error screen should then be presented. Follow the instruction to install Woo, if the setup succeeds, you should be navigated to the home screen.

Please feel free to double check that the new button doesn't appear on the site picker when there is at least one site found.

### Screenshots

Troubleshooting a self-hosted site without Jetpack and WooCommerce:

https://user-images.githubusercontent.com/5533851/185050209-bcad3cf7-d612-4a60-9ac6-30a693b2e216.mp4



---
